src/librustc/lint/builtin.rs

```diff
@@ -386,6 +386,12 @@ declare_lint! {
     "ambiguous associated items"
 }
 
+declare_lint! {
+    pub REDUNDANT_IMPORT,
+    Warn,
+    "redundant import"
+}
```

Comment:
I think this can be merged into `unused_imports` without creating a new lint, even if it does a slightly different thing.
FWIW, redundant `extern crate`s are also reported by `unused_extern_crates`.

Reply:
Ok, I look into it